.github/scripts/change-workflow-instance-states.py

```diff
@@ -6,17 +6,24 @@
 
 import argparse
 
-from common import change_workflow_instance_states
+from platform_lib import Platform, get_platform_enum
+from common import get_platform_lib
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser()
+    parser.add_argument('platform',
+                        choices = ['aws', 'azure', 'all'],
+                        help = "The platform CI is being run on")
     parser.add_argument('tag_value',
                        help = "The tag used to identify workflow instances.")
     parser.add_argument('state_change',
                        choices = ['terminate', 'stop', 'start'],
                        help = "The state transition to initiate on workflow instances.")
     parser.add_argument('github_api_token',
                        help = "API token to modify self-hosted runner state.")
-
     args = parser.parse_args()
-    change_workflow_instance_states(args.github_api_token, args.tag_value, args.state_change)
+    platform = get_platform_enum(args.platform)
+    if platform == Platform.AWS or platform == Platform.ALL:
```

Comment:
When do you expect to use `Platform.ALL`? I wonder if we can just remove this functionality / potential confusion.

Reply:
Internally I use it in my `setup-self-hosted-runner` job to catch the failed instances (since I setup both the Azure / AWS runner in the same step in the workflow). This just prevents duplication of the that step in the job. Ostensibly I can also just do this in 2 separate steps to avoid this, but that leads to a lot more copy-pasta. There are some small things like this that can probably be cleaned up after I get around to workflow generation like you suggested, but I need to get that working first.